tests/app/views/test_brief_response.py

```diff
@@ -607,6 +738,103 @@ def test_can_not_submit_an_invalid_brief_response(
             }
         }
 
+    def test_can_not_submit_brief_response_with_non_legacy_data(self, live_dos_framework):
+        # To create a brief_resonse with an invalid key from the new schema, we need to switch the feature flag on.
+        # We switch it back off after creation to allow us to exhibit the behaviour we're testing.
+        feature_flag_date = (datetime.utcnow() - timedelta(days=7)).strftime("%Y-%m-%d")
+        self.app.config["FEATURE_FLAGS_NEW_SUPPLIER_FLOW"] = feature_flag_date
+
+        non_legacy_brief_response_data = {
+            'essentialRequirementsMet': True,
+            'niceToHaveRequirements': [True, False, True, False, True],
+            'availability': u'a',
+            'respondToEmailAddress': 'supplier@email.com'
+        }
+
+        create_res = self.create_brief_response(data=non_legacy_brief_response_data)
+        assert create_res.status_code == 201
+
+        brief_response_id = json.loads(create_res.get_data(as_text=True))['briefResponses']['id']
+
+        self.app.config["FEATURE_FLAGS_NEW_SUPPLIER_FLOW"] = self.feature_flag_date
```

Comment:
Could we mark this with a comment? I know you mentioned it above (a duplicate of the last sentence from that would do) but it would really help the reading to have that info next to the code.